src/lib/markbind/src/lib/markdown-it/index.js

```diff
@@ -45,17 +45,44 @@ markdownIt.renderer.rules.fence = (tokens, idx, options, env, slf) => {
   if (!highlighted) {
     str = markdownIt.utils.escapeHtml(str);
   }
+  let highlightLines = token.attrGet('highlight-lines')
+  let lineNumbers = []
+  if (highlightLines) {
+    lineNumbers = highlightLines.split(',').map(v => v.split('-').map(v => parseInt(v, 10))) 
+  }
+
   const lines = str.split('\n');
   lines.pop(); // last line is always a single '\n' newline, so we remove it
-  str =  lines.map(line => `<span>${line || '&#x200B;'}</span>`).join('');
+  str =  lines.map((line, index) => {
+    const currentLineNumber = index + 1
+    const inRange = lineNumbers.some(([start, end]) =>{
+      if (start && end) {
+        return currentLineNumber >= start && currentLineNumber <= end
+      }
+      return currentLineNumber === start
+    })
+    if (inRange) {
+      return `<span class="highlighted">${line || '&#x200B;'}</span>`;
+    }
+    return `<span>${line || '&#x200B;'}</span>`;
+  }).join('')
   // wrap all lines with <span> so we can number them
   // if a line is empty we put a 0 width non breaking space
   token.attrJoin('class', 'hljs');
 
   if (highlighted) {
     token.attrJoin('class', lang);
   }
-  return `<pre><code ${slf.renderAttrs(token)}>${str}</code></pre>`;
+
+  const heading = token.attrGet('heading')
+  if (heading) {
+    return `<div class='code-block'>`
+      + `<div class='code-block-heading'><span>` + heading + `<span></div>` 
+      + `<div class='code-block-content'><pre><code ${slf.renderAttrs(token)}>${str}</code></pre></div>`
```

Comment:
Same as above, lets abstract the common part into it's own variable.

Reply:
Done